src/basic/label.h

```diff
@@ -0,0 +1,14 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include <sys/types.h>
+
+typedef struct LabelOps {
+        int (*pre)(int dir_fd, const char *path, mode_t mode);
+        int (*post)(int dir_fd, const char *path);
```

Comment:
(we should eventually add another entry fix() here, that relabels things. And then strip direct selinux support from most of our codebase, and only go via labelops)